packages/iTwinUI-react/src/core/utils/hooks/useDragAndDrop.tsx

```diff
@@ -0,0 +1,71 @@
+/*---------------------------------------------------------------------------------------------
+ * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+ * See LICENSE.md in the project root for license terms and full copyright notice.
+ *--------------------------------------------------------------------------------------------*/
+import React from 'react';
+
+/**
+ * Helper hook that handles elements drag logic.
+ * @param elementRef Element ref that is draggable.
+ * @returns
+ * `onPointerDown` - handler that is called when pointer is down and handles all the dragging logic.
+ * `transform` - current transform of the element, it is used to preserve drag position when element visibility is being toggled.
+ */
+export const useDragAndDrop = (elementRef: React.RefObject<HTMLElement>) => {
+  const grabOffsetX = React.useRef(0);
+  const grabOffsetY = React.useRef(0);
+  const translateX = React.useRef(0);
+  const translateY = React.useRef(0);
+
+  const [transform, setTransform] = React.useState('');
+
+  const onPointerMove = React.useCallback(
+    (event: PointerEvent) => {
```

Comment:
why here `event: PointerEvent` and below `e: React.PointerEvent<HTMLElement>` ?

Reply:
Because `onPointerMove` is native event handler (added to the document) and `onPointerDown` is React event handler.